src/compton2/Compton2.cc

```diff
@@ -0,0 +1,475 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   compton2/Compton2.cc
+ * \author Andrew Till
+ * \date   11 May 2020
+ * \brief  Implementation file for compton interface
+ * \note   Copyright (C) 2017-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+// headers provided in draco:
+#include "compton2/Compton2.hh"
+#include "c4/C4_Functions.hh" //node()
+#include "c4/global.hh"
+#include "ds++/Assert.hh"
+
+#include <algorithm>
+#include <array>
+#include <cstring>
+#include <fstream>
+#include <iomanip> // RM
+#include <ios>     // RM
+#include <iostream>
+
+using UINT = size_t;
+using FP = double;
+using vec = std::vector<FP>;
+
+namespace rtt_compton2 {
+
+// xs is assumed to be monotonically increasing and unique
+// returns index in [0,len-2] so that xs[index] and xs[index+1] are valid
+// modifies x so that xs[index] <= x <= xs[index+1]
+UINT find_index(const std::vector<FP> &xs, FP &x) {
+  // Do binary search for index
+  auto loc = std::upper_bound(xs.begin(), xs.end(), x);
+  // loc - begin guaranteed to be in [0,len]
+  UINT index = (loc - xs.cbegin());
+
+  // Move index and x to be interior
+  // len >=2 assumed
+  UINT len = xs.size();
+  FP xmin = xs[0];
+  FP xmax = xs[len - 1U];
+  index = std::min(len - 1U, std::max(UINT(1), index)) - 1U;
+  x = std::max(xmin, std::min(xmax, x));
+  return index;
+}
+
+// xL <= x <= xR assumed
+std::array<FP, 4> hermite(FP x, FP xL, FP xR) {
+  // Spacing of interval
+  FP h = xR - xL;
+  // Left/right basis functions
+  FP bL = (xR - x) / h;
+  FP bR = (x - xL) / h;
+
+  // Hermite functions
+  std::array<FP, 4> H{bL * bL * (3 - 2 * bL), bR * bR * (3 - 2 * bR), -h * bL * bL * (bL - 1),
+                      h * bR * bR * (bR - 1)};
+  return H;
+}
+
+Compton2::Compton2(std::string filename)
```

Comment:
Add Class doc block.

Reply:
I'll dig around for one. Is anyone using Doxygen these days?

Reply:
* Try using one from `ds++/DracoStrings.hh` 
* Something like:
```c++
//------------------------------------------------------------------------------------------------//
/*!
 * \brief one-liner
 *
 * \param[in] string_in This string will be converted letter by letter to lowercase.
 * \return A string that contains no uppercase letters.
 *
 * Longer description as needed, with design, intent, possibly examples.
 */
```